prdoc/pr_4035.prdoc

```diff
@@ -0,0 +1,21 @@
+title: "Prospective parachains rework"
+
+doc:
+  - audience: Node Dev
+    description: |
+      Reworks prospective-parachains so that we allow a number of unconnected candidates (for which we don't yet know
+      the parent candidate). Needed for elastic scaling. Also simplifies it to not allow parachain forks and cycles.
```

Comment:
I'd first mention that this fundamentally changes what information the subsystem stores and operates on. From a tree to just a chain and bunch of unconnected candidates.

Reply:
reworked the prdoc, let me know if it sounds better now